pkg/scanner/scan.go

```diff
@@ -66,21 +71,30 @@ func ScanImage(imageName, filePath string, scanOptions types.ScanOptions) (map[s
 		if err != nil {
 			return nil, xerrors.Errorf("failed to scan libraries: %w", err)
 		}
+
+		var libResults report.Results
 		for path, vulns := range libVulns {
-			results[path] = vulns
+			libResults = append(libResults, &report.Result{
+				FileName:        path,
+				Vulnerabilities: vulns,
+			})
 		}
+		sort.Slice(libResults, func(i, j int) bool {
+			return libResults[i].FileName < libResults[j].FileName
+		})
+		results = append(results, libResults...)
 	}
 
 	return results, nil
 }
 
-func ScanFile(f *os.File) (map[string][]vulnerability.DetectedVulnerability, error) {
+func ScanFile(f *os.File) (report.Results, error) {
```

Comment:
This function doesn't seem to be used anywhere. Should this be removed?

Reply:
Yes, but this function is used by a different tool.
https://github.com/knqyf263/remic/blob/master/remic.go#L69

Before, I'd like to develop a different tool to scan lock files. I suspend to develop it now, but I'm supposed to resume it in the future. Let me keep this function.